src/azure/azureController.ts

```diff
@@ -279,4 +318,15 @@ export class AzureController {
 		}
 	}
 
+	public set SubscriptionClient(v: SubscriptionClient) {
```

Comment:
What is this for? We don't want to use the same client for the entire azureController - many different things can call into that

Reply:
It seems like this is for testing purposes. It's much better to use the DI pattern where you inject a factory method into the constructor - something like this : 

```
export type SubscriptionClientFactory = (token: string) => string;
function defaultSubscriptionClientFactory(token: string) {
	return new SubscriptionClient(new TokenCredentialWrapper(token));
}
...
constructor(private _subscriptionClientFactory: SubscriptionClientFactory = defaultSubscriptionClientFactory) {
```

And then whenever you need to get the client you just call `this._subscriptionClientFactory(token)`

This removes a lot of the "extra" code and is a lot easier to understand since you're using the factory in the same way regardless of how it was implemented.